src/diagnostics.rs

```diff
@@ -36,8 +35,11 @@ pub struct DiagnosticReport {
     pub os_name: String,
     pub os_version: String,
     pub triple: String,
+    pub is_ci: bool,
     pub action: DiagnosticAction,
     pub status: DiagnosticStatus,
+    /// Generally this includes the [`strum::IntoStaticStr`] representation of the error, we take special care not to include parameters of the error (which may include secrets)
+    pub variant: Option<String>,
```

Comment:
We discussed renaming this so its meaning is more obvious when looking through the diagnostics:

```suggestion
    pub failure_variant: Option<String>,
```